pkg/yttlibrary/assert.go

```diff
@@ -25,6 +26,27 @@ var (
 
 type assertModule struct{}
 
+func (b assertModule) Equals(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 2 {
+		return starlark.None, fmt.Errorf("expected two arguments")
+	}
+
+	expected, err := core.NewStarlarkValue(args.Index(0)).AsGoValue()
```

Comment:
Since we don't want to allow lambda functions as arguments, we will need to explicitly check for them prior to trying to convert them to a go value (otherwise we get an 'error converting function to go value'). We could do a check similar to 

https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/yttlibrary/assert.go#L46-L49

But instead error if we get a `starlark.Callable` type.